src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rest/QueryBatchRequestHandler.java

```diff
@@ -0,0 +1,309 @@
+      return;
+    }
+
+    exchange.getResponseHeaders().set("Content-Type", "application/json");
+
+    Map<String, String> params = getParamsMap(exchange.getRequestURI().getQuery());
+
+    try {
+      // Parse and validate parameters
+      String[] validParamsTmp = {"", "metrics", "starttime", "endtime", "samplingperiod"};
+      Set<String> validParams = new HashSet<>(Arrays.asList(validParamsTmp));
+      for (String param : params.keySet()) {
+        if (!validParams.contains(param)) {
+          throw new InvalidParameterException(String.format("%s is an invalid parameter", param));
+        }
+      }
+
+      List<String> metrics = metricsRestUtil.parseArrayParam(params, "metrics", false);
+      String startTimeParam = params.get("starttime");
+      String endTimeParam = params.get("endtime");
+      String samplingPeriodParam = params.get("samplingperiod");
+
+      for (String metric : metrics) {
+        if (!MetricsModel.ALL_METRICS.containsKey(metric)) {
+          throw new InvalidParameterException(String.format("%s is an invalid metric", metric));
+        }
+      }
+
+      if (startTimeParam == null || startTimeParam.isEmpty()) {
+        throw new InvalidParameterException("starttime parameter must be set");
+      }
+      long startTime;
+      try {
+        startTime = Long.parseUnsignedLong(startTimeParam);
+      } catch (NumberFormatException e) {
+        throw new InvalidParameterException(String.format("%s is an invalid starttime", startTimeParam));
+      }
+
+      if (endTimeParam == null || endTimeParam.isEmpty()) {
+        throw new InvalidParameterException("endtime parameter must be set");
+      }
+      long endTime;
+      try {
+        endTime = Long.parseUnsignedLong(endTimeParam);
+      } catch (NumberFormatException e) {
+        throw new InvalidParameterException(String.format("%s is an invalid endtime", endTimeParam));
+      }
+
+      long samplingPeriod = DEFAULT_SAMPLING_PERIOD_MILLIS;
+      if (samplingPeriodParam != null && !samplingPeriodParam.isEmpty()) {
+        samplingPeriod = Long.parseLong(samplingPeriodParam);
+        if (samplingPeriod < 5 || samplingPeriod % 5 != 0) {
+          throw new InvalidParameterException(String.format("%s is an invalid sampling period", samplingPeriodParam));
+        }
+        if (samplingPeriod >= PluginSettings.instance().getBatchMetricsRetentionPeriodMinutes() * 60) {
+          throw new InvalidParameterException("sampling period must be less than the retention period");
+        }
+        samplingPeriod *= 1000;
+      }
+
+      if (startTime >= endTime) {
+        throw new InvalidParameterException("starttime must be less than the endtime");
+      }
+      startTime -= startTime % samplingPeriod;
+      endTime -= endTime % samplingPeriod;
+      if (startTime == endTime) {
+        throw new InvalidParameterException("starttime and endtime cannot be equal when rounded down to the nearest sampling period");
```

Comment:
nit: should we reword as something on the lines of - "Start time and end time should be at least sampling period apart".? It may be confusing in its current form.

Reply:
Agreed